src/Email/PHPMailer.php

```diff
@@ -67,6 +68,17 @@ public function send(bool $debug = false): bool
             $mailer->Port       = $this->transport()['port'] ?? null;
         }
 
+        // accessible phpMailer instance
+        $callback = $this->callback();
+
+        if (empty($callback) === false && is_a($callback, 'Closure')) {
+            $mailer = $callback->call($this, $mailer);
+
+            if (empty($mailer) === true || is_a($mailer, 'PHPMailer\PHPMailer\PHPMailer') === false) {
+                throw new InvalidArgumentException('"callback" option return should be instance of PHPMailer\PHPMailer\PHPMailer');
+            }
```

Comment:
As the `$mailer` object is passed by reference to the function, the return value should be optional:

- If the return value is `null`, the original `$mailer` value should be used.
- If the return value is anything else, the object type should be validated like in your implementation.